enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,320 @@
+
+The HyperShift's API which will handle the hosted cluster capabilities is out of the scope of this enhancement. However this enhancement is built with the assumption that the `ClusterVersion`'s capability API (spec and status) is sufficient for any capability implementation in HyperShift.
+
+#### Standalone OpenShift
+
+Making the ingress operator optional for the standalone OpenShift is a bigger scope than for HyperShift.
+The changes impact the components which are not used (or used differently) in HyperShift like the installer, the authentication and the monitoring operators.
+To prevent the scope creep for [RFE-3395](https://issues.redhat.com/browse/RFE-3395), the following approach is proposed:
+- the cluster version operator should be able to add required capabilities.
+- the required capabilities is a configuration option of the cluster version operator (e.g. `--required-capabilities` flag).
+- the cluster version operator should [implicitly enable](https://github.com/openshift/cluster-version-operator/blob/2fd54d18de83b617d9d30c28cfee6383e430a102/pkg/cvo/status.go#L161-L165) all the provided required capabilities so that they appear in the status of `ClusterVersion` object.
+- the required capabilities should be added to [the cluster version operator manifests](https://github.com/openshift/cluster-version-operator/blob/master/install/0000_00_cluster-version-operator_03_deployment.yaml) only.
+- the ingress capability should be required.
+
+This approach should make any required capability read only on the standalone OpenShift.
+Also, it doesn't let capabilities be set as required on HyperShift because HyperShift doesn't use the cluster version operator's deployment manifest but [manages the operator's deployment directly](https://github.com/openshift/hypershift/blob/15fe51bda33b7acede2b4b7a7c66acc2f369a8a2/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L318),
+so the required capabilities list argument won't be supplied on HyperShift.
+
+#### Ingress operator APIs
+
+The following APIs are not available if the ingress capability is disabled:
+- CRDs which are part of the cluster operator's manifest payload:
+  - [IngressController](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/00-custom-resource-definition.yaml)
+  - [DNSRecord](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/00-custom-resource-definition-internal.yaml)
+- CRDs which are created at runtime if `gateway-api` feature gate is set:
+  - [GatewayClass](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_gatewayclasses.yaml)
+  - [Gateway](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_gateways.yaml)
+  - [HTTPRoute](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_httproutes.yaml)
```

Comment:
* Might as well use the active voice to make it explicit that the ingress operator is the thing that normally creates these CRDs.
* The feature gate is named "GatewayAPI", not "gateway-api".  
* Might as well mention referencegrants for completeness.  
* Use permalinks when possible.  
```suggestion
- CRDs which the cluster ingress operator creates at runtime if the `GatewayAPI` feature gate is enabled:
  - [GatewayClass](https://github.com/openshift/cluster-ingress-operator/blob/d319e461f1c74a80474804a447237a8a679c4abd/assets/gateway-api/gateway.networking.k8s.io_gatewayclasses.yaml)
  - [Gateway](https://github.com/openshift/cluster-ingress-operator/blob/d319e461f1c74a80474804a447237a8a679c4abd/assets/gateway-api/gateway.networking.k8s.io_gateways.yaml)
  - [HTTPRoute](https://github.com/openshift/cluster-ingress-operator/blob/d319e461f1c74a80474804a447237a8a679c4abd/assets/gateway-api/gateway.networking.k8s.io_httproutes.yaml)
  - [ReferenceGrant](https://github.com/openshift/cluster-ingress-operator/blob/d319e461f1c74a80474804a447237a8a679c4abd/assets/gateway-api/gateway.networking.k8s.io_referencegrants.yaml)
```